lib/AST/LocalArchetypeRequirementCollector.cpp

```diff
@@ -183,13 +183,29 @@ Type MapLocalArchetypesOutOfContext::getInterfaceType(
     ++depth;
   }
 
+  // If the environment we have is not for a capture, this must be a generic
+  // environment local to the parent function. For element archetypes, the depth
+  // we have is relative to the parent signature, so we can compute the correct
+  // depth by offsetting it by the number of parent captured environments.
```

Comment:
This might not be correct if you have multiple nested `for` loops over packs. What causes us to end up in this code path?

Reply:
This handles e.g:

```swift
func foo<each T>(_ xs: repeat each T) {
  for x in repeat each xs {}
}
```

There is no captured environment for `x`, it's local to the function itself. Multiple nested should be fine since we track the parent generic signature in the statement checker, let me update the test case

Reply:
Oh hrm, looks like it doesn't work in a closure 🤔 Seems like we don't pass down the innermost element environment

Reply:
That isn't really an issue with this patch though, this crashes on main:

```swift
func foo<each T>(xs: repeat each T) {
  _ = {
    for x in repeat each xs {
      for y in repeat each xs {
        return (x, y)
      }
    }
  }
}
```